.github/workflows/build_matrix.py

```diff
@@ -0,0 +1,45 @@
+import json
+import sys
+
+# list of images/python combinations can be found here:
+# https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
+python_version = ["3.8", "3.9", "3.10", "3.11"]
+
+# None indicates to skip that combination
+ubuntu_version = ["ubuntu-22.04", "ubuntu-22.04", "ubuntu-24.04", "ubuntu-24.04"]
+macos_version = ["macos-latest", "macos-latest", "macos-latest", "macos-latest"]
+windows_version = [None, None, "windows-latest", "windows-latest"]
+
+# installation, minimal version of python
+packages = [
+    ("'numpy~=1.24.0' 'scipy<1.15.0'", "3.8"),
```

Comment:
Python 3.8 is unsupported since 2024-10-07, it might not be worth spending CI on testing it.

Reply:
I have no strong opinions about this and I'll follow the trend. I'll remove it when we agree to merge this.